src/nodes/Functions/NaturalLog.ts

```diff
@@ -0,0 +1,36 @@
+import Decimal from 'decimal.js';
+import TreeNode, { ITreeNode } from '../TreeNode';
+
+/**
+ * NaturalLog node
+ * @class NaturalLog
+ */
+
+export default class NaturalLog extends TreeNode implements ITreeNode {
+  operand : TreeNode;
+
+  constructor(...x : TreeNode[]) {
+    if (x.length !== 1) {
+      throw new SyntaxError(`ln takes exactly one argument, received ${x.length}`);
+    }
+    super('ln');
+    [this.operand] = x;
+  }
+
+  /**
+   * Evaluates the natural log function
+   * @returns the result of the natural log function
+   */
+  evaluate() {
+    const x = new Decimal(this.operand.evaluate());
```

Comment:
There is no need to convert to `Decimal` here, `Decimal.ln` works with arguments of type string